[FLINK-30064] Move existing Hive connector code from Flink repo to dedicated Hive repo

snuyanzin opened the pull request:
The PR is to move hive connector to external repo

history was taken based on https://cwiki.apache.org/confluence/display/FLINK/Externalized+Connector+development#ExternalizedConnectordevelopment-Githistory

luoyuxia committed:
[FLINK-23995][hive] Fix bug of table name parser when using Hive dialect

This closes #16999

RyanSkraba committed:
[FLINK-24476][docs] Rename Elasticsearch case in docs. This closes #17451

Commit:
[FLINK-24940][docs] Correct usage about how to create Hive catalog via Flink SQL CLI. This closes #17829

xiangqiao123 committed:
[FLINK-25142][hive] Fix user-defined hive UDTF initialize exception in hive dialect

This closes #19400

zentol committed:
[FLINK-25868][build] Disable japicmp in packaging/e2e modules

Optimize build times and reduce noise (warnings would be printed as these modules are not being deployed, so the previous versions can not be found).

Commit:
[FLINK-26681][hive] Support sql end with ";" for Hive dialect

This closes #19105

MartijnVisser committed:
[FLINK-27063][Connectors/Hive] Upgrade Hive 2.3 connector from 2.3.6 to 2.3.9

Commit:
[FLINK-27116][hive] Support read Hive table partitioned by decimal type

This closes #19388

zentol committed:
[FLINK-27125][tests] Fix ExecutorService leaks

twalthr committed:
[hotfix][table-planner] Simplify PlannerContext signatures related to CatalogManager

twalthr committed:
[FLINK-22857][table-planner] Simplify DefaultSqlExprToRexConverterFactory to RexFactory

Commit:
[FLINK-27217][hive] Support partition filter push down when there exists default partition for Hive source.

This closes #19450

zentol committed:
[FLINK-27252][hive][build] Remove surefire fork settings

Commit:
[FLINK-27244][hive] Support read sub-directories in partition directory with Hive tables

This closes #19482

Commit:
[FLINK-26366][hive] Support INSERT OVERWRITE DIRECTORY statement

This closes #19520

yangjf2019 committed:
[FLINK-23902][connectors/hive] Add support to Hive version 3.1.3.

This closes #19512.

zentol committed:
[FLINK-24721][build] Remove unnecessary relativePath to parent

Commit:
[FLINK-27304][hive] Calcite's varbinary type should be converted to Hive's binary type.

This closes #19562

zentol committed:
[FLINK-27453][tests] Cleanup TestBaseUtils

alpreu committed:
[FLINK-27185][connector] Convert connector-hive module to assertj

Co-authored-by: slinkydeveloper <francescoguard@gmail.com>

Tartarus0zm committed:
[FLINK-27450][hive] Upgrade default hive version to fix Hive SessionState initialization issue with jdk11

snuyanzin committed:
[FLINK-27323][table][tests] Migrate table-api-java to JUnit5

luoyuxia committed:
[FLINK-15854][hive] Use the new type inference for Hive UDTF (#18958)

luoyuxia committed:
[FLINK-28036][hive] HiveInspectors should use correct writable type to create ConstantObjectInspector

This closes #19949

luoyuxia committed:
[FLINK-27999][hive] Fix Hive3 issues

- VirtualColumn.VIRTUAL_COLUMN_NAMES has a different type in hive2 vs hive3, causing NoSuchFieldErrors since our code is only compiled against hive2
- HiveConf is a singleton that is modified in various ways; adds a band-aid to resolve some issues

luoyuxia committed:
[FLINK-27895][hive][ci] Enable nightly Hive 3.x tests

zentol committed:
[FLINK-28203][build] Mark bundled dependencies as optional

luoyuxia committed:
[FLINK-28011][hive] HiveSource should get partitions in batch instead of getting partition one by one

This closes #20015

luoyuxia committed:
[FLINK-28247][hive] fix "no any field presented in group by" exception when over window contains grouping function in Hive dialect

This closes #20071.

swuferhong committed:
[FLINK-27988][table-planner] Let HiveTableSource extend from SupportStatisticsReport

This closes #20084

luoyuxia committed:
[FLINK-26361][hive] Create LogicalFilter with CorrelationId to fix failed to rewrite subquery in hive dialect (#18920)

Tartarus0zm committed:
[FLINK-25142][hive] (followups) Fix user-defined hive UDTF initialize exception in hive dialect

This closes #19400

JingsongLi committed:
[FLINK-28322][table-api] DataStreamScan(Sink)Provider's new method is not compatible

This closes #20119

luoyuxia committed:
[FLINK-15855][hive] Use the new type inference for Hive UDAF

This closes #19920

wuchong committed:
[FLINK-27244][hive] Improve documentation of reading partition with subdirectories for Hive tables

luoyuxia committed:
[FLINK-26414][hive] Hive dialect supports macro (#19561)

luoyuxia committed:
[FLINK-28451][table][hive] Use UserCodeClassloader instead of the current thread's classloader to load function

This closes #20211

eyys committed:
[FLINK-28471][docs-zh] Translate hive_read_write.md into Chinese

This closes #20227

luoyuxia committed:
[FLINK-26412][hive] Hive dialect supports "CREATE FUNCTION ... USING JAR" statement

This closes #20251

luoyuxia committed:
[FLINK-28525][hive] Fix unstable test for HiveDialectITCase#testTableWithSubDirsInPartitionDir

This closes #20259.

luoyuxia committed:
[FLINK-27376][table] Support current_database built-in function (#19218)

luoyuxia committed:
[FLINK-28772][hive] Supports ADD JAR command in Hive dialect

This closes #20413

fsk119 committed:
[FLINK-28150][sql-gateway][hive] Introduce the hiveserver2 endpoint and factory

This closes #20101

luoyuxia committed:
[FLINK-27013][hive] Hive dialect supports IS_DISTINCT_FROM (i.e. "<=>" operator)

This closes #19315

luoyuxia committed:
[FLINK-28089][hive] Hive dialect support "TABLESAMPLE (N rows)" (#19989)

luoyuxia committed:
[FLINK-27996][hive] Hive dialect supports INTERVAL types

This closes #19955

wuchong committed:
[FLINK-28451][hive] Borrow Kryo from SerializationUtilities in HiveFunctionWrapper

This closes #20211

fsk119 committed:
[FLINK-28152][sql-gateway] Simplify SessionHandle and OperationHandle

fsk119 committed:
[FLINK-28152][sql-gateway][hive] Allow executing statement and fetching results for the HiveServer2Endpoint

fsk119 committed:
[FLINK-28152][sql-gateway][hive] Support GetOperationStatus and GetResultSetMetadata for HiveServer2Endpoint

libenchao committed:
[FLINK-28676] Update copyright year to 2014-2022 in NOTICE files

fsk119 committed:
[FLINK-28152][sql-gateway][hive] Introduce option "thrift.host" for HiveServer2 Endpoint and improve codes

luoyuxia committed:
[FLINK-28628][table] Introduce Operation Execution Plugin (#20247)

luoyuxia committed:
[FLINK-27338][hive] Improve splitting file for Hive source

This closes #20377

empcl committed:
[FLINK-27384][hive] Fix the modified partitions are missed in temporal table with create-time mode

This closes #20376.

fsk119 committed:
[FLINK-28679][hive] HiveServer2 Endpoint supports to build with Hive3 (#20358)

luoyuxia committed:
[FLINK-27387][hive] Hive dialect supports multi-insert statement (#19647)

luoyuxia committed:
[FLINK-26410][hive] Hive dialect supports "transform using 'xxx'"

This closes #19818

fsk119 committed:
[FLINK-27770][sql-gateway] Refactor SqlGateway E2E tests

This closes #20174

luoyuxia committed:
[FLINK-28096][hive] Hive dialect support set variables (#20000)

chaozwn committed:
[FLINK-28151][hive] Allow to cancel the Operation for the HiveServer2 Endpoint.

This closes #20402

luoyuxia committed:
[FLINK-26542][hive] Hive dialect supports queries with multiple same columns in select (#19017)

fsk119 committed:
[FLINK-28629][sql-gateway][hive] Allow to getCatalogs in the HiveServer2 Endpoint

This closes #20334

luoyuxia committed:
[FLINK-27016][hive] Support access array type in a struct and support access fields in a struct array

This clsoes #19283

Co-authored-by: Jark Wu <jark@apache.org>

pltbkd committed:
[FLINK-28711][hive] Hive source supports dynamic filtering

This closes #20415

chaozwn committed:
[FLINK-28793][sql-gateway][hive] Allow to GetInfo in the HiveServer2 Endpoint

This closes #20444

luoyuxia committed:
[FLINK-28212][hive-dialect] Fix IndexOutOfBoundsException when over window SELECT doesn't contain all fields of input

This closes #20060

yuzelin committed:
[FLINK-28630][sql-gateway][hive] Allow to GetSchemas in the HiveServer2 Endpoint

This closes #20401

Tartarus0zm committed:
[FLINK-28720][hive][connectors] Add Hive partition when flink has no data to write (#20394)

luoyuxia committed:
[FLINK-28795][hive] Fix unstable test for HiveDialectQueryITCase#testInsertDirectory (#20446)

chaozwn committed:
[FLINK-28851][hive] Allow to GetTypeInfo in the HiveServer2 Endpoint

This closes #20468

luoyuxia committed:
[FLINK-26371][hive] Support variable substitution for sql statement while using Hive dialect (#19656)

luoyuxia committed:
[FLINK-26362][hive] Fix IndexOutOfBoundsException when sub-query selects all fields  when using Hive dialect

This closes #18937

wuchong committed:
[FLINK-27217][hive] Improve default partition test cases

This closes #19450

luoyuxia committed:
[FLINK-28296][hive] Support the Hive UDAF which implement GenericUDAFResolver interface (#20102)

yuzelin committed:
[FLINK-28633][sql-gateway][hive] Allow to GetTables in the HiveServer2 Endpoint

This closes #20438

Tartarus0zm committed:
[FLINK-28773][hive] Allow to write a success file after finish for Hive sink in batch mode

This closes #20469

wuchong committed:
[FLINK-28772][hive] Improve exception messages for ADD JAR

This closes #20413

fsk119 committed:
[FLINK-28631][sql-gateway][hive] Support to GetFunctions in the HiveServer2Endpoint

This closes #20479

luoyuxia committed:
[FLINK-28797][hive] HiveSource enables vector reading for complex data type with parquet format

fsk119 committed:
[FLINK-28851][sql-gateway][hive] Remove useless OperationType

yuzelin committed:
[FLINK-28632][sql-gateway][hive] Allow to GetColumns/GetPrimaryKeys/GetTableTypes in the HiveServer2 Endpoint

This closes #20493

luoyuxia committed:
[FLINK-26413][hive] Supports "LOAD DATA INPATH" in Hive dialect

This closes #19556

wuchong committed:
[FLINK-28797][hive] Simplify useModules for the parquet complex type tests

luoyuxia committed:
[FLINK-28778][SQL/API] Bulk fetch of table and column statistics for given partitions

This closes #20501

Co-authored-by: Jing Ge <gejing@gmail.com>

luoyuxia committed:
[FLINK-28774][hive] Allow user to configure whether to enable sort not when it's for dynamic partition writing for HiveSource

This closes #20419

twalthr committed:
[FLINK-28861][table] Make UID generation behavior configurable and plan-only by default

Before this commit, due to changes for FLIP-190, every operator generated by the planner
got a UID assigned. However, the UID is based on a static counter that might return different
results depending on the environment. Thus, UIDs are not deterministic and make stateful
restores impossible e.g. when going from 1.15.0 -> 1.15.1. This PR restores the old pre-1.15
behavior for regular Table API. It only adds UIDs if the operator has been created from a
compiled plan. A compiled plan makes the UIDs static and thus deterministic.

table.exec.uid.generation=ALWAYS exists for backwards compatibility and could make stateful
upgrades possible even with invalid UIDs on best effort basis.

luoyuxia committed:
[FLINK-26771][hive] Fix incomparable exception between boolean type and numeric type in Hive dialect

This closes #19182

yuzelin committed:
[FLINK-28855][hive] Fix 'Table.INDEX_TABLE' not found in the Hive3 (#20514)

luoyuxia committed:
[FLINK-28795][hive] Fix unstable test in HiveDialectQueryITCase

This closes #20569.

Myracle committed:
[FLINK-28208][hive] Set parallelism for map operator in class HiveTableSink's method createBatchSink (#20471)

luoyuxia committed:
[FLINK-28913][hive] Fix failed to open HiveCatalog when it's for hive3 (#20573)

wuchong committed:
[FLINK-28773][hive] Improve tests of writing success files

This closes #20469

swuferhong committed:
[FLINK-28992][table-planner] Fix: Change Ndv takes the max value instead of sum of all partitions when getting partition table column stats

This closes #20595

luoyuxia committed:
[FLINK-27386][hive] Fix assert exception when specific join hint in union statement (#19583)

luoyuxia committed:
[FLINK-26505][hive] Support non equality condition for left semi join in Hive dialect. (#18994)

luoyuxia committed:
[FLINK-27015][hive] Fix exception for casting timestamp to decimal in Hive dialect (#20571)

pltbkd committed:
[FLINK-28995][hive] HiveSourceDynamicFileEnumerator filters out a partition without matching if one of the fields is null but its type is not nullable

This closes #20612

Tartarus0zm committed:
[FLINK-28965][hive] Partition shouldn't be created if no data is generated for dynamic partition

This closes #20599

luoyuxia committed:
[FLINK-28956][hive] Fix non-multi insert statement fall into multi insert logic in Hive dialect (#20572)

swuferhong committed:
[FLINK-29046][Connectors/Hive] Fix HiveTableSourceStatisticsReportTest fails with Hive 3.x

This closes #20664

swuferhong committed:
[FLINK-28993][table-planner] Refactor SupportsDynamicFiltering to provide getAcceptedFilterFields method to avoid modifing the DynamicTableSource object if calling applyDynamicFiltering method in join reorder rules

This closes #20596

luoyuxia committed:
[FLINK-28938][hive] Fix HiveServer2 Endpoint can not set variable correctly

luoyuxia committed:
[FLINK-27175][hive] Fix fail to call Hive UDAF when the UDAF accepts one parameter with array type

This closes #19423

luoyuxia committed:
[FLINK-28883][hive] Fix HiveTableSink failed to report statistic to hive metastore in batch mode

This closes #20549

luoyuxia committed:
[FLINK-29126][hive] Fix splitting file optimization doesn't work for orc format

This closes #20694

luoyuxia committed:
[FLINK-29025][docs] add overview page for Hive dialect

fsk119 committed:
[FLINK-28938][hive] Improve error messages for unsupported interfaces

fsk119 committed:
[FLINK-29148][docs][sql-gateway][hive] Add SQL Gateway docs

This closes #20719

yuchuanchen committed:
[FLINK-29005][parquet] Parquet row type reader should not return null value when some child fields is null

This closes #20616

luoyuxia committed:
[FLINK-26474][hive] Fold exprNode to fix the issue of failing to call some hive udf required constant parameters with implicit constant passed

This closes #18975

yuzelin committed:
[FLINK-29118][sql-gateway][hive] Remove default GenericInMemoryCatalog in the HiveServer2 Endpoint when openSession

 This closes #20714

snuyanzin committed:
[hotfix][tests] Replace deprecated AbstractThrowableAssert#getRootCause

HuangXingBo committed:
Update version to 1.17-SNAPSHOT

fsk119 committed:
[FLINK-29188][sql-gateway][hive] Make option "sql-gateway.endpint.hiveserver2.catalog.hive-conf-dir" required

luoyuxia committed:
[FLINK-27017][hive] Fix divide by zero exception with Hive dialect (#19216)

fsk119 committed:
[hotfix] Bind to broadcast address when host is not specified

luoyuxia committed:
[FLINK-29013][hive] Fix fail to use BinaryRecordReader in "transform using" syntax with Hive dialect (#20643)

luoyuxia committed:
[FLINK-27718][hive] Fix fail to count mutiple fields excpetion in Hive dialect (#19406)

luoyuxia committed:
[FLINK-29185][hive] Fix ClassNotFoundException for CREATE TEMPORARY FUNCTION USING JAR with Hive dialect

This closes #20776

fsk119 committed:
[FLINK-29228][hive] Align the schema of the HiveServer2 getMetadata with JDBC

fsk119 committed:
[FLINK-29229][hive] Fix HiveServer2 endpoint doesn't support execute statements in sync mode

zentol committed:
[FLINK-29211][hive][legal] Update 2.3.9 NOTICE

wuchong committed:
[FLINK-29025][docs][hive] Fix links of Hive compatibility pages

luoyuxia committed:
[FLINK-29222][hive] Fix wrong behavior for Hive's load data inpath

This closes #20778

luoyuxia committed:
[FLINK-29191][hive] Fix Hive dialect can't get value for the variables set by set command

This closes #20774

luoyuxia committed:
[FLINK-29152][hive] Fix "SHOW CREATE TABLE" doesn't work for Hive dialect

This closes #20795

luoyuxia committed:
[FLINK-29152][hive] Fix inconsistent behavior with Hive for `desc table` in Hive dialect 

This closes #20789

luoyuxia committed:
[FLINK-29386][hive] Fix failed to compile flink-connector-hive in hive3 profile (#20882)

fsk119 committed:
[FLINK-29274][hive] Fix ObjectStore leak when different users has different config (#20891)

lsyldliu committed:
[FLINK-29219][table] Fix CREATE TABLE AS statement blocks SQL client's execution

This closes #20869

pltbkd committed:
[FLINK-29348][table] DynamicFilteringEvent can be stored in the CoordinatorStore if it's received before the listener source coordinator is started.

This closes #20906.

snuyanzin committed:
[FLINK-20873][table] Upgrade Calcite to version 1.27

This closes #20850.

luoyuxia committed:
[FLINK-29408][hive][tests] Do not move resource

luoyuxia committed:
[FLINK-29337][hive] Fix fail to query non-hive table in Hive dialect 

This closes #20855

zoucao committed:
[FLINK-26726][connector][hive]Hive enumerators do not assign splits to unregistered (failed) readers

This closes #19250

MartijnVisser committed:
[FLINK-29624][Common][Connector][Filesystem] Upgrade org.apache.commons:commons-lang3 from 3.3.2 to 3.12.0

* Remove Pulsar specific version of `commons-lang3` since Flink now uses a newer version
* Replace usage of deprecated method `org.apache.commons.lang3.ArrayUtils#add` with `org.apache.commons.lang3.ArrayUtils#insert`
* Replace usage of `org.apache.commons.lang3.StringEscapeUtils` with `org.apache.commons.text.StringEscapeUtils`

luoyuxia committed:
[FLINK-29350][hive] Add a section for moving planner jar in Hive dependencies page

This closes #20913

luoyuxia committed:
[FLINK-29590][hive] Fix literal issue in Hive dialect (#21026)

Samrat002 committed:
[FLINK-29478][Connector/Hive] Upgrade Hive connector to 3.1.3. This closes #20937

MartijnVisser committed:
[FLINK-29710][Filesystem] Bump minimum supported Hadoop version to 2.10.2

- Make the SQL Client use the Hadoop versioning as defined in the parent POM
- Make the SQL Gateway use the Hive and Hadoop versioning as defined in the parent POM
- Move and clarify Hive specific Hadoop versioning in `flink-connector-hive`
- Sync stax2-api to solve dependency convergence in hadoop-common
- Sync commons-beanutils and stax2-api to solve dependency convergence in hadoop-common
- Sync commons-beanutils and stax2-api to solve dependency convergence in hadoop-common
- Fix YarnTestBase to work with new Hadoop version. Also remove comment on issue with previously supported Hadoop version
- Disable HDFS Client to remove a failed datanode and to never add a new datanode when an existing one is removed. Its recommended to disable this for small clusters, which is the case in our tests
- Bump Hadoop 3 to 3.2.3.

HADOOP-12984 changed where the MiniYARNCluster wrote to by default.

2.10.2: https://github.com/apache/hadoop/blob/965fd380006fa78b2315668fbc7eb432e1d8200f/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-tests/src/test/java/org/apache/hadoop/yarn/server/MiniYARNCluster.java#L177

This hasn't made it's way into 3.1.3: https://github.com/apache/hadoop/blob/aa96f1871bfd858f9bac59cf2a81ec470da649af/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-tests/src/test/java/org/apache/hadoop/yarn/server/MiniYARNCluster.java#L175

But was added to 3.2.3 in MAPREDUCE-7320.

Co-authored-by: Gabor Somogyi <gabor_somogyi2@apple.com>
Co-authored-by: Chesnay Schepler <chesnay@apache.org>

reswqa committed:
[hotfix] fix typo in HiveOptions.

HuangXingBo committed:
[FLINK-29733][hive] Fix the version error in HiveVersionTestUtil

zentol committed:
[hotfix] Remove deprecated Assertj API usages

yuzelin committed:
[FLINK-29857][hive] Set default value for GetInfoResp

luoyuxia committed:
[FLINK-29831][hive] fix test failure for hive3

Commit:
[FLINK-29878][hive] Fix flink-sql-connector hive error

MartijnVisser committed:
[FLINK-29710][Hadoop/Hive] Exclude Reload4J from all Hadoop and Hive dependencies

- Include an update to the bannedDependency enforcer rule for Log4J1 to also include Reload4J

luoyuxia committed:
[FLINK-29992][hive] Fix Hive lookup join fail when column pushdown to Hive lookup table source

This closes ##21302.

vvcephei committed:
[FLINK-29962] Exclude jamon 2.3.1 from dependencies (#21278)

MartijnVisser committed:
[FLINK-30170][Parquet] Update Parquet to 1.12.3

lsyldliu committed:
[FLINK-29718][table] Supports hive sum function by native implementation

This closes #21401

snuyanzin committed:
[hotfix] replace deprecated calcite apis

snuyanzin committed:
[hotfix] Replace deprecated Calcite's APIs (#21459)

* RexWindowBound#create and RexWinAggCall
* RelCollationImpl#of
* AggregateCall#create
* SqlCharStringLiteral#getNlsString

Commit:
[FLINK-30019][hive] Remove the unused HiveTableFileInputFormat in hive connector

This closes #21316

liyubin117 committed:
[FLINK-29679][table] Migrate to new schema framework & show column comment

This closes #21182

luoyuxia committed:
[FLINK-30452][hive] fix use wrong argument type of null literal is used when call Hive's function

This closes #21528

MartijnVisser committed:
[FLINK-27640][Connector/Hive][SQL Client] Exclude Pentaho dependency from Hive to avoid build problems with newer Maven versions. Newer Maven versions block http repositories such as conjars. Conjars is where Pentaho artifacts are stored

lsyldliu committed:
[FLINK-30491][hive] Hive table partition supports lazy deserialization during runtime

This closes #21556

fsk119 committed:
[FLINK-30493][table-api] Introduce TableChange to SET/RESET options (#21554)

This closes #21554

fsk119 committed:
[FLINK-30496][table-api] Introduce TableChange to represent MODIFY change

This closes #21577

lsyldliu committed:
[FLINK-29719][hive] Supports native count function for hive dialect

This closes #21596

lsyldliu committed:
[FLINK-27920][hive] Supports native avg function for hive dialect

This closes #21601

fsk119 committed:
[FLINK-30497][table-api] Introduce TableChange to represent DROP change

This closes #21592

lsyldliu committed:
[FLINK-29722][hive] Supports native hive max function for hive dialect

This closes #21605

Tartarus0zm committed:
[FLINK-29721][hive] Supports native hive min function for hive dialect

This closes #21608

chrismartin823 committed:
[FLINK-30592][doc] Remove unsupported hive version in hive overview document (#21611)

lsyldliu committed:
[FLINK-30579][hive] Introducing configurable option to use hive native function

This closes #21629

lincoln-lil committed:
[FLINK-30006][table-planner] Make all pure dynamic functions non-deterministic in streaming mode to avoid wrong plans

luoyuxia committed:
[FLINK-30617][hive] Fix wrong data type for cast null literal with Hive dialect

This closes #21642

MartijnVisser committed:
[hotfix] Rename incorrect reference to Hive specific Hadoop version

yuzelin committed:
[FLINK-29950][sql-gateway] Refactor ResultSet to an interface 

This closes #21502

chenqin committed:
[FLINK-27640][Connector/Hive] Exclude pentaho-aggdesigner-algorithm from Hive connector dependencies. This closes #21653

LadyForest committed:
[FLINK-19883][table] Support "IF EXISTS" in DDL for ALTER TABLE 

This closes #21640

yuzelin committed:
[FLINK-30693][sql-gateway] Supports to specify RowFormat when fetching results

This closes #21677

JunRuiLee committed:
[FLINK-30683][runtime] Use AdaptiveBatchScheduler as default batch scheduler

This closes #21672.

luoyuxia committed:
[FLINK-29878][hive] Include libfb303 to fix fail to use flink-sql-connector-hive-3.1.3

fsk119 committed:
[FLINK-29945][sql-client] Supports to submit SQL to an embedded SQL Gateway in the SQL Client

This closes #21717 

Co-authored-by: Jane Chan <55568005+LadyForest@users.noreply.github.com>

JunRuiLee committed:
[FLINK-30684][runtime] Mark vertices which use the default parallelism

For these vertices, the parallelism can be changed at runtime.

This closes #21772.

luoyuxia committed:
[FLINK-30567][hive] Fix wrong behavior for insert overwrite with Hive dialect when the table contains uppercase character  (#21602)

[FLINK-30567][hive] Fix wrong behavior for insert overwrite with Hive dialect when the table contains uppercase character 

This closes #21602

JunRuiLee committed:
[FLINK-30685][runtime] Mark transformations of which the parallelism is infected by input transformations

For these transformations, the parallelism can be changed at runtime.

This closes #21780.

hehuiyuan committed:
[FLINK-30679][hive] Fix IndexOutOfBoundsException for Hive lookup join when column pushdown to Hive lookup table source #21731

Co-authored-by: hehuiyuan1 <hehuiyuan@jd.com>

JunRuiLee committed:
[FLINK-30686][conf] Refine adaptive batch configurations

This closes #21787.

fsk119 committed:
[FLINK-30692][sql-client] Introduce SingleSessionManager to support REMOVE JAR syntax (#21771)

luoyuxia committed:
[FLINK-29880][hive] Introduce auto compaction for Hive sink in batch mode (#21703)

LadyForest committed:
[FLINK-30386] Fix the issue that column constraint lacks primary key not enforced check

This closes #21507

leonardBang committed:
Update version to 1.18-SNAPSHOT

shuiqiangchen committed:
[FLINK-31124][Connectors/Hive] Add IT case for HiveTableSink speculative execution

This closes #21962.

lsyldliu committed:
[FLINK-30824][hive] Add document for option 'table.exec.hive.native-agg-function.enabled'

This closes #21789

reswqa committed:
[FLINK-31132][hive] Let hive compact operator without setting parallelism subject to sink operator's configured parallelism.

This closes #21977

reswqa committed:
[hotfix] Fix typo in HiveTableSink and HiveTableCompactSinkITCase.

reswqa committed:
[hotfix][test] Migrate HiveTableCompactSinkTICase, HiveTableFactoryTest, HiveTableSinkITCase to Junit5 and AssertJ.

fsk119 committed:
[FLINK-31137][hive] Fix wrong ResultKind in DescribeTable and ShowCreateTable results (#21980)

fsk119 committed:
[FLINK-31175][doc] Fix link doesn't work in hive_read_write doc (#21988)

lsyldliu committed:
[FLINK-31239][hive] Fix native sum function can't get the corrected value when the argument type is string

This closes #22031

lsyldliu committed:
[FLINK-31239][hive] Add some notes in document about table.exec.hive.native-agg-function.enabled option can't turned on per job when using it via SqlClient

This closes #22031

luoyuxia committed:
[FLINK-31409][table] Introduce public interfaces for pluggable dialect

luoyuxia committed:
[FLINK-31409][hive] Hive dialect shouldn't use interfaces in table-planner to build LogicPlan
This closes #22166

luoyuxia committed:
[FLINK-30659][hive] Drop flink-sql-parser-hive module (#22106)

* [FLINK-30659][hive] Drop flink-sql-parser-hive module

snuyanzin committed:
[FLINK-31369][sql-gateway] Harden modifiers in tests for sql-gateway module

Commit:
[FLINK-29585][hive] Migrate TableSchema to Schema for Hive connector (#21522)

lincoln-lil committed:
[FLINK-31487][table-planner] Add targetColumns to DynamicTableSink#Context

This closes #22197

rmetzger committed:
[hotfix] Update copyright NOTICE year to 2023

luoyuxia committed:
[FLINK-31531][hive] Fix HiveCatalogHiveMetadataTest&HiveDialectITCase failure in Hive3 (#22230)

TanYuxin-tyx committed:
[FLINK-30815][tests] Migrate BatchAbstractTestBase to junit5

This closes #22427

zentol committed:
[FLINK-31804] Allow MiniClusterTestEnvironment

zentol committed:
[hotfix][hive] Mark mapreduce-client-core as provided

Clarify that this dependency is meant to be provided by the user.

luoyuxia committed:
[FLINK-31413][hive] Hive connector shouldn't use PlannerQueryOperation

luoyuxia committed:
[FLINK-31413][hive] Hive connector shouldn't use OperationConverterUtils

luoyuxia committed:
[FLINK-31413][hive] Remove usage org.apache.flink.table.planner.functions.* of in Hive connector

luoyuxia committed:
[FLINK-31413][hive] Copy HintPredicate,NodeTypeHintPredicate, LogicalDistribution,LogicalScriptTransform,ExpressionBuilder
from table-planner to Hive connector

luoyuxia committed:
[FLINK-31413][hive] HiveParser allows to parse Flink's special commands

luoyuxia committed:
[FLINK-31413][hive] Change scope of flink-table-planner dependency from provided to test in Hive connector
This closes #22620

luoyuxia committed:
[FLINK-30660][e2e] Add e2e test for hive

rbowen committed:
[docs] Deduplicate mistakenly repeated words

luoyuxia committed:
[FLINK-32294][e2e] Fix HiveITCase failure (#22740)

StefanRRichter committed:
[FLINK-32410] Replace new HashMap<>(size) with CollectionUtil.newHashMapWithExpectedSize(size)

StefanRRichter committed:
[FLINK-32410] Replace new HashSet<>(size) with CollectionUtil.newHashSetWithExpectedSize(size)

StefanRRichter committed:
[FLINK-32410] Replace new LinkedHashMap<>(size) with CollectionUtil.newLinkedHashMapWithExpectedSize(size)

MartijnVisser committed:
[FLINK-32032] Upgrade to flink-shaded 17.0. This closes #22546

* Upgrade to flink-shaded 17.0
* Replace all `guava30` occurrences with `guava31` since flink-shaded uses a newer version of Guava
* Upgrade Curator to 5.4.0
* Update the flink-connector-jdbc as used by `flink-python` to 3.1.1-1.17 since that version doesn't rely on Flink Shaded
* Use a specific `flink-shaded-guava` version in the Confluent Schema Registry test to avoid ClassNotFound errors

snuyanzin committed:
[FLINK-32526] Bump Apache Parquet from 1.12.3 to 1.13.1

Commit:
[FLINK-32388][filesystem]Add the ability to pass parameters to CUSTOM PartitionCommitPolicy (#22831)

Baibaiwuguo committed:
[FLINK-32365][hive] Allow Hive source to scan table to get statistics in parallel (#22805)

reswqa committed:
[FLINK-32620][sink] Migrate DiscardingSink to sinkv2.

luoyuxia committed:
[FLINK-31575][hive] Avoid swapping table-planner-loader and table-planner to use hive dialect (#22862)

jiaoqingbo committed:
[FLINK-32223][runtime][security] Add Hive delegation token support

XComp committed:
Revert "[FLINK-32223][runtime][security] Add Hive delegation token support"

This reverts commit c30057b6f889d98c06fc04a1c9f913d8e4cbb9d8.

It was decided to revert this issue due to compilation errors in hadoop_313 CI stage. See the Jira issue for further details.

LadyForest committed:
[FLINK-32656][table] Deprecate ManagedTable related APIs

This closes #23065.

fsk119 committed:
[FLINK-32731][e2e] Backup the hive-metastore, hdfs namenode and mysql logs (#23155)

JingGe committed:
Update version to 1.19-SNAPSHOT

dongwoo6kim committed:
[FLINK-32760][Connectors/Hive] Reshade shaded parquet in flink-sql-connector-hive (#23166)

fsk119 committed:
[FLINK-32731][e2e] Add retry mechanism when fails to start the namenode (#23267)

luoyuxia committed:
[FLINK-32982][doc] Not to suggest to swap flink-table-planner lib for using Hive dialect
This closes #23309

fsk119 committed:
[FLINK-32731][e2e] Fix NameNode uses port 9870 in hadoop3

This closes #23370

WencongLiu committed:
[FLINK-32978][flink-core] Migrate the usage of RichFunction#open(Configuration parameters) to RichFunction#open(OpenContext openContext)

This closes #23058

Jiabao-Sun committed:
[FLINK-32997][table-planner][JUnit5 Migration] Module: flink-connectors (StreamingTestBase)

Jiabao-Sun committed:
[FLINK-33023][table-planner][JUnit5 Migration] Module: flink-connectors/flink-connector-hive (TableTestBase)

jiaoqingbo committed:
[FLINK-32223][runtime][security] Add Hive delegation token support

swuferhong committed:
[FLINK-25593][table-planner] Skip redundant scan while partition table push down contains none-existent partition (#23423)

yunfan123 committed:
[FLINK-33233][hive] Fix NPE when non-native udf is used in join condition with Hive dialect

ferenc-csaky committed:
[FLINK-33226][table] Forbid to drop current database

XComp committed:
[FLINK-33418][test] Uses getHost()

yunfan123 committed:
[FLINK-33478][hive] Enable foldExpr by default in filter condition (#23682)

snuyanzin committed:
[FLINK-30064][hive] Move Hive connector to external repo

Jiabao-Sun committed:
[FLINK-33551][Connectors/Hive] Fix HiveTemporalJoinITCase.testProcTimeTemporalJoinHiveTable nightly CI fails

MartijnVisser commented on .github/workflows/push_pr.yml:
```diff
@@ -25,4 +25,4 @@ jobs:
     uses: apache/flink-connector-shared-utils/.github/workflows/ci.yml@ci_utils
     with:
-      flink_version: 1.16.0
+      flink_version: 1.19-SNAPSHOT
```
Is the externalization based on the code from the `release-1.18` branch or master?

snuyanzin replied:
initial commit is based on master however let's see if there could be done support for 1.18.0
there are several things which are required to be workaround like FLINK-32620, FLINK-25593 and some others

snuyanzin replied:
yes, with last 2 commits 1.18 is supported.

there could be 2 ways: support it for 1.18.0 first only and then apply similar approach to adapt updates from 1.19
or support it for both 1.19 and 1.18  like here

snuyanzin committed:
[hotfix] Remove Calcite classes from Hive connector

snuyanzin commented:
rebased since there were new commits in main flink master to hive-connector area

Jiabao-Sun committed:
[FLINK-33637][table-planner][JUnit5 Migration] Introduce ArchTest to ban Junit4 for table-planner module

This closes #23791.

RyanSkraba reviewed (commented):
Hello Sergey -- sorry this took so long to review!  In principle, it looks pretty good, but I have a question about the migration of test utilities from 

* flink: `./flink-end-to-end-tests/flink-end-to-end-tests-common/src/main` to
* flink-connector-hive: `/flink-connector-hive-e2e-tests/src/main/`

1. The migrated package is `org.apache.flink.tests.util.flink` but other connectors use the form `org.apache.link.tests.util.kafka` for example.
2. Just for info, are these utilities unique to the flink Hive connector?

There's some weirdness going on with my local build, where I can't seem  to get the docker images in the TestContainers to run, but it's hard to argue with the CI passing!  I'm investigating the reason locally.

Thanks for this work!!
These files are necessary for the flink end-to-end test

JingGe committed:
[hotfix][connector-hive][test] fix ArchUnit typo in pom

luoyuxia commented:
Cool! @snuyanzin  Thanks for the pr. Sorry for the late response for I'm busy with other things. I'll definitely have a look in next week.

snuyanzin commented:
rebased since there were new commits in main flink master to hive-connector area

also added archunit violations because of recent commit within FLINK-33637
@RyanSkraba moved test packages to `flink.connectors.hive`

luoyuxia commented on flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/HiveTableSinkITCase.java:
```diff
@@ -0,0 +1,1000 @@
+            final TableEnvironment tEnv, final String expectedResourceFileName) throws Exception {
+        tEnv.registerCatalog(hiveCatalog.getName(), hiveCatalog);
+        tEnv.useCatalog(hiveCatalog.getName());
+        if (FlinkVersion.current() == FlinkVersion.v1_18) {
```
nit: 
Seems like there some code ducplicated in here and in line [154](https://github.com/apache/flink-connector-hive/pull/5/files#diff-d854ec6577a7ee748c98aef66fc72a22f8059929c8c2851be8d2c7ba234d6803R154). Can they share a common code fragment.

snuyanzin replied:
good catch, thanks
done

luoyuxia reviewed (commented):
@snuyanzin Thanks for the great work. Overall look good to me  assuming ci test pass for Hive2 & Hive3.

luoyuxia commented:
@snuyanzin Hi, I also found some pages(https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/hive-compatibility/hive-dialect/overview/ , https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql-gateway/hiveserver2/) in Flink still refer to Hive with internal link. Will we adjust them in the pr of removing Hive connector from flink repo?

snuyanzin commented:
Thanks for the feedback
> Will we adjust them in the pr of removing Hive connector from flink repo?

yep, usually it's happening either with while removal of connector from main repo or together with the first oficial release of externalised connector.

MartijnVisser reviewed (approved):
LGTM

luoyuxia reviewed (approved):
@snuyanzin Thanks for the update. LGTM. Btw, will we also enable test with hive3 nightly just like flink does? I want to make sure we can have a chance to  test & verify it with hive3.

MartijnVisser commented:
> I want to make sure we can have a chance to test & verify it with hive3.

Good point. I actually think we should consider running them both for each PR, that should be easier with Hive being externalized.

snuyanzin commented:
agree, makes sense
will try do that
however currently this is no supported by [flink-connector-shared-utils](https://github.com/apache/flink-connector-shared-utils) ...

the only solution is to copy paste `ci.yml` and enable running tests against different hive versions

I will see if it could be supported on a generic way since same functionality would be nice to have for OpenSearch connector

MartijnVisser commented:
> I will see if it could be supported on a generic way since same functionality would be nice to have for OpenSearch connector

We could consider adding `optional_maven_profiles` as a generic variable and pass that through (kind of like https://github.com/apache/flink-connector-shared-utils/blob/ci_utils/.github/workflows/ci.yml#L87-L89)

snuyanzin commented:
yep, that exactly the idea i'm testing right now =)
thanks for confirmation :+1:

snuyanzin commented:
just another note, I noticed that hive3 is tested only for jdk8 in flink nightlies
I tried to run it with jdk 11 and e2e failed while for hive2 it is passing
Since it was not run with jdk11 i will add support only for jdk8 for hive3 as it is done in flink main repo
other jdk support could be done separately

snuyanzin committed:
[FLINK-33718][table] Cleanup the usage of deprecated StreamTableEnvironment#toAppendStream

snuyanzin commented:
Currently it works for hive2 and hive3 (without e2e for jdk11)
as

|hive version   |    flink 1.18 + jdk 8      |  flink 1.19-snapshot +  jdk8 |      flink 1.18 + jdk 11      |  flink 1.19-snapshot +  jdk11   |
|----------|:-------------:|------:|------------------------------:|---------------------:|
| 2 | ✅ | ✅ |✅ |✅ |
|3 |   ✅ |✅ | no E2E  | no E2E|

as mentioned above under flink main there is no E2E tests for jdk11 as well

The PR currently copies `ci.yaml` from [flink-connector-shared-utils](https://github.com/apache/flink-connector-shared-utils) ...

at the same time there are 2 PRs for that repo to support customization of jdk version https://github.com/apache/flink-connector-shared-utils/pull/24 and maven profiles https://github.com/apache/flink-connector-shared-utils/pull/29 , these 2 features are used here. Once they are merged and a new release released  `ci.yml` could be reused and the copy could be removed

@luoyuxia , @MartijnVisser please let me know whether you are ok with this or not

luoyuxia commented:
@snuyanzin Cool! Thanks for your work. It sounds good to me.

MartijnVisser commented:
> please let me know whether you are ok with this or not

I'm reviewing the CI PRs; when they are merged and we can merge this, will we remove the Hive code from Flink's `master` branch?

snuyanzin commented:
>I'm reviewing the CI PRs; when they are merged and we can merge this

thanks a lot for your review

> will we remove the Hive code from Flink's master branch?

yes, that was the plan, the only question is whether we need a dedicated ML thread for that or since there is already existing jira issue for Hive externalisation, we could just continue with it?

Also I would guess we need to release this connector

MartijnVisser commented:
> dedicated ML thread for that or since there is already existing jira issue for Hive externalisation, we could just continue with it?

It would be good to announce the intent at least, to also avoid that Hive commits get merged into `master` between the time the release (which we should make indeed) is completed and the externalization happening

snuyanzin committed:
[hotfix] Allow manual trigger for weekly

snuyanzin commented:
After merging of mentioned PRs to [flink-connector-shared-utils](https://github.com/apache/flink-connector-shared-utils) all related WAs are removed and `ci_utils` is reused.

Also there are a couple of commits to hive connector in main repo, one of which leaded to conflicts. Now they are integrated here as well

The PR also enables manual trigger of nightly however it will be possible to trigger this only after merge to main
I will trigger it on my own fork

I'm going to merge it soon after the build is green and nightly  on my fork is also green

snuyanzin commented:
weekly passed 
https://github.com/snuyanzin/flink-connector-hive/actions/runs/7145518824
